apps/backend/test/mocks/mocks.cash.spec.ts

```diff
@@ -31,4 +32,25 @@ describe('Tests the generated mock data', () => {
       expected.deposits.map((itm) => itm.amount)
     );
   });
+
+  it('should generate cash deposits with a date greater than the date of the last payment', () => {
+    const expected = {
+      deposits: cashDepositsMock.map((itm) => ({
+        date: itm.deposit_date,
+        amount: itm.deposit_amt_cdn
+      })),
+      payments: aggregatedPayments(cashPaymentsMock).map((itm) => ({
+        date: itm.fiscal_close_date,
+        amount: itm.amount
+      }))
+    };
+    expected.deposits.forEach((deposit) =>
+      expected.payments.forEach((payment) =>
+        expect(
```

Comment:
So I don't love this test, and initially I thought it was because it wasn't obvious what the expected value is supposed to look like, but ultimately, it's supposed to just be a bunch of 1s. So I think that's fine.

But I think what makes it less readable / maybe what I don't like about it is that it's unclear how many expects it is gonna go through. The double array iteration means it's running x*y times, which could be a lot, or could be 1 or 4 or 9.

I'm not sure what the solution is though. Could it be pulling some random indexes and comparing those? Or pulling some specific ones?

I guess my other question is, do we HAVE to test, e.g. deposits[2] with payments[0], payments[1], payments[2], payments[3] ...? That's currently what's happening right? Or do we just want to test deposits[2] with just payments[2]?

Reply:
This one is just meant to test that the mock data is accurate - this particular test on line 47 is only meant to verify that the payments mock fiscal close date is earlier than the cash deposit_date, since deposits always occur after payments.

Reply:
If the first date in the comparison (the deposit date) is before the second date in the comparison (the payment date), then the output from calling 'compareAsc' is -1, and the test will fail. If the dates are equal, then the output from calling 'compareAsc' is 0, and the test will fail. If the first date(the deposit date) is after the second date(the payment date) then calling 'compareAsc' will return with a 1, and the test will pass.

...probably a better way to check this...

Reply:
(I deleted my initial response after a re-read. I am going to a refactor on this....all payment dates should be earlier than all deposit dates, so I think there's an easier way to do this.)